packages/mui-base/src/utils/useComponentRenderer.ts

```diff
@@ -63,6 +65,8 @@ export function useComponentRenderer<
     customStyleHookMapping,
   } = settings;
 
+  const children = extraProps?.children;
```

Comment:
Temporary. The proper solution will require an explicit `children` parameter (preferably required TS-wise when `render === 'asChild'`).